pkgs/development/compilers/gcc/ng/common/gcc/default.nix

```diff
@@ -0,0 +1,246 @@
+{
+  lib,
+  stdenv,
+  gcc_meta,
+  release_version,
+  version,
+  monorepoSrc ? null,
+  langAda ? false,
+  langC ? true,
+  langCC ? true,
+  langD ? false,
+  langFortran ? false,
+  langGo ? false,
+  langJava ? false,
+  langObjC ? stdenv.targetPlatform.isDarwin,
+  langObjCpp ? stdenv.targetPlatform.isDarwin,
+  langJit ? false,
+  enablePlugin ? lib.systems.equals stdenv.hostPlatform stdenv.buildPlatform,
+  runCommand,
+  buildPackages,
+  isl,
+  zlib,
+  gmp,
+  libmpc,
+  mpfr,
+  libelf,
+  perl,
+  texinfo,
+  which,
+  gettext,
+  getVersionFile,
+  buildGccPackages,
+  targetPackages,
+  libc,
+  bintools,
+}:
+let
+  inherit (stdenv) targetPlatform hostPlatform;
+  targetPrefix = lib.optionalString (targetPlatform != hostPlatform) "${targetPlatform.config}-";
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "${targetPrefix}${if langFortran then "gfortran" else "gcc"}";
+  inherit version;
+
+  src = monorepoSrc;
+
+  outputs = [
+    "out"
+    "man"
+    "info"
+  ];
+
+  patches = [
+    # Submitted:
+    #  - https://gcc.gnu.org/pipermail/gcc-patches/2021-August/577639.html
+    #  - https://gcc.gnu.org/pipermail/gcc-patches/2021-August/577640.html
+    #  - https://gcc.gnu.org/pipermail/gcc-patches/2021-August/577638.html
+    #
+    # In Git: https://github.com/Ericson2314/gcc/tree/prog-target-14
+    #
+    # This is all 3 commits squashed.
+    (getVersionFile "gcc/find-prefixed-progs.patch")
```

Comment:
Ditto what I said about splitting back up.

Reply:
Do I just need to squash your fixup patches commit with the init commit?

Reply:
Yup!